### PR TITLE
Add removeGlobalEventJIP

### PR DESCRIPTION
**Pull request by PabstMirror**
Idea is from something @Dystopian did in https://github.com/acemod/ACE3/pull/5418
Adds a way to remove things from the globalEventJip queue.
Either immedietly or if passed an object it will remove the jipEvent if object is deleted.

`remoteExec` has something similar and it can be set to not run on jips if object is deleted.

Doesn't cause problems, but one issue is that `setVar [x, nil]` does not remove from `allVariables`
But it still stops the event from running.

### CODE REVIEW

**Comment by commy2** [comment]
>Doesn't cause problems, but one issue is that setVar [x, nil] does not remove from allVariables

-_-

File: addons/events/fnc_removeGlobalEventJIP.sqf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Author:
+    PabstMirror (idea from Dystopian)
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+SCRIPT(removeGlobalEventJIP);
+
+params [["_jipID", "", [""]], ["_object", nil, [nil, objNull]]];
+
+if ((isNil "_object") || {isNull _object}) then {
```
Comment (commy2) [inline addons/events/fnc_removeGlobalEventJIP.sqf]:
```sqf
params [["_jipID", "", [""]], ["_object", objNull, [objNull]]];

if (isNull _object) then {
```

**Comment by Dystopian** [comment]
call looks strange (to me):
```
_jipID = ["event"] call CBA_fnc_globalEventJIP;
[_jipID, _object] call CBA_fnc_removeGlobalEventJIP;
```
I mean we add the event and then remove it. Maybe function name should differ, e.g. `unsetGlobalEventJIP`.

File: addons/events/fnc_removeGlobalEventJIP.sqf [hunk 2]
```diff
@@ -6,7 +6,7 @@ Description:
 
 Parameters:
     _jipID  - A unique ID from CBA_fnc_globalEventJIP. <STRING>
-    _object - Object, will remove jip EH when object is deleted. Or pass nil to ignore and remove immediately [optional] <OBJECT>or<NIL>
+    _object - Will remove jip EH when object is deleted or immediately if object is null [optional] <OBJECT>
```
Comment (Dystopian) [inline addons/events/fnc_removeGlobalEventJIP.sqf]: or immediately if omitted

**Comment by commy2** [comment]
```sqf
execVM "\x\cba\addons\events\test_globalEventJIP.sqf"
```
```
10:13:13 [CBA] (events) Test OK: (CBA_fnc_globalEventJIP is defined) x\cba\addons\events\test_globalEventJIP.sqf:13
10:13:13 [CBA] (events) Test OK: (CBA_fnc_removeGlobalEventJIP is defined) x\cba\addons\events\test_globalEventJIP.sqf:14
10:13:13 [CBA] (events) Test OK: (cba_events_test_A == _expected) Verify EH ran x\cba\addons\events\test_globalEventJIP.sqf:23
10:13:13 [CBA] (events) Test OK: (not (isNil {cba_events_eventNamespaceJIP getVariable _ret})) x\cba\addons\events\test_globalEventJIP.sqf:24
10:13:13 [CBA] (events) Test OK: (isNil {cba_events_eventNamespaceJIP getVariable _ret}) x\cba\addons\events\test_globalEventJIP.sqf:28
10:13:13 [CBA] (events) Test OK: (isNil {cba_events_eventNamespaceJIP getVariable _ret}) x\cba\addons\events\test_globalEventJIP.sqf:33
10:13:13 [CBA] (events) Test OK: (not (isNil {cba_events_eventNamespaceJIP getVariable _ret})) x\cba\addons\events\test_globalEventJIP.sqf:39
10:13:15 [CBA] (events) Test OK: (isNull _dummyObject) x\cba\addons\events\test_globalEventJIP.sqf:42
10:13:15 [CBA] (events) Test OK: (isNil {cba_events_eventNamespaceJIP getVariable _ret}) x\cba\addons\events\test_globalEventJIP.sqf:43
```

**Pull request merged**